sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -335,6 +358,19 @@ trait Row extends Serializable {
   def getAs[T](fieldName: String): T = getAs[T](fieldIndex(fieldName))
 
   /**
+   * Returns the value of a given fieldName as a scala.util.Try object.
+   */
+  def attempt[T](fieldName: String): Try[T] = Try(getAs[T](fieldIndex(fieldName)))
```

Comment:
Thanks for responding @srowen. I have not used the Java API for Spark. But if I understand what you said correctly, then yes, one of the uses for `getOption` was to get `None`s where the value is `null`. However, I also wanted exceptions to be converted to `None`s. I believe the use case is common enough that we can have a method for it, instead of calling `Try(getAs[Foo]("bar")).toOption` every time we want an `Option`al value. What do you think?